ext/http/lib.rs

```diff
@@ -425,6 +428,39 @@ impl Default for HttpResponseWriter {
   }
 }
 
+struct BodyCompressedHolder {
+  pub writer: Pin<Box<dyn tokio::io::AsyncWrite>>,
+  pub shutdown_handle: ShutdownHandle,
+}
+
+struct BodyUncompressedSender {
+  sender: Option<hyper::body::Sender>,
+  done: bool,
```

Comment:
What's the reason for having this as a separate `bool`, rather than just `.take()`ing the stream once done?

Reply:
Yes, good point. I restructured the code to get rid of it.